File: src/main/webapp/resources/js/customer/role/roleService.js
//服务层
app.service('roleService',function($http){
	this.findAll=function(){
		return $http.get('../../../role/findAll');
	}
	this.findAllCotainSuperAdm=function(){
		return $http.get('../../../role/findAllCotainSuperAdm');
	}
	this.findAllT=function(id){
		return $http.get('../../role/findAll');
	}
	this.findById=function(id){
		return $http.get('../../../role/findById?id='+id);
	}
	this.update=function(roleEntity){
		return $http.post('../../role/update',roleEntity);
	}
	this.add=function(roleEntity){
		return $http.post('../../role/add',roleEntity);
	}
	this.del=function(ids){
		return $http.get('../../role/del?ids='+ids);
	}
	this.resourceTree=function(roleId){
		return $http.get('../../role/resourceTree?roleId='+roleId);
	}
	this.grantPermission=function(roleId,resourceIds){
		return $http.get('../../../role/grantPermission?roleId='+roleId+"&resourceIds="+resourceIds);
	}
});

File: src/main/java/com/webside/role/service/RoleService.java
package com.webside.role.service;

import java.util.List;
import java.util.Map;

import org.apache.ibatis.annotations.Param;

import com.webside.plateformaccount.model.UserEntity;
import com.webside.role.model.RoleEntity;
import com.webside.util.JsonResult;
import com.webside.util.PageData;
import com.webside.util.PageResult;

public interface RoleService {
    
    public List<RoleEntity> findAll(PageData pd);
    public PageResult<List<RoleEntity>> findPage(PageData pd);
    public RoleEntity findByName(String name);
    public void add(RoleEntity roleEntity);
    public RoleEntity findById(Integer id);
    public void update(RoleEntity roleEntity);
    public JsonResult delete(int[] ids);
    public boolean addRolePermBatch(int roleId, List<Integer> ids);
    public boolean addRolePerm(Integer roleId, Long resourceId);
}